presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveRedirectionToIceberg.java

```diff
@@ -0,0 +1,129 @@
+        extends HiveProductTest
+{
+    private static final String CREATE_ICEBERG_TABLE_TEAMPLATE = "CREATE TABLE %s (_string VARCHAR, _bigint BIGINT, _date DATE) WITH (partitioning = ARRAY['_bigint', '_date'])";
+    private static final String CREATE_HIVE_TABLE_TEAMPLATE = "CREATE TABLE %s (_string VARCHAR, _bigint BIGINT, _date DATE) WITH (partitioned_by = ARRAY['_bigint', '_date'])";
+    private static final String INSERT_TEMPLATE = "INSERT INTO %s VALUES " +
+            "(NULL, NULL, NULL), " +
+            "('abc', 1, DATE('2020-08-04')), " +
+            "('abcdefghijklmnopqrstuvwxyz', 2, DATE('2020-08-04'))";
+    private static final List<Row> EXPECTED_ROWS = ImmutableList.<Row>builder()
+            .add(row(null, null, null))
+            .add(row("abc", 1, Date.valueOf("2020-08-04")))
+            .add(row("abcdefghijklmnopqrstuvwxyz", 2, Date.valueOf("2020-08-04")))
+            .build();
+
+    @Test(groups = {HIVE_REDIRECTION_TO_ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testSelectAndDrop()
+    {
+        onPresto().executeQuery(format(CREATE_ICEBERG_TABLE_TEAMPLATE, "iceberg.default.test_select_drop"));
+        onPresto().executeQuery(format(INSERT_TEMPLATE, "iceberg.default.test_select_drop"));
+        assertThat(onPresto().executeQuery("SELECT * FROM iceberg.default.test_select_drop")).containsOnly(EXPECTED_ROWS);
+        assertThat(onPresto().executeQuery("SELECT * FROM hive.default.test_select_drop")).containsOnly(EXPECTED_ROWS);
+        assertThat(onPresto().executeQuery("SELECT _bigint, _date FROM hive.default.\"test_select_drop$partitions\"")).containsOnly(
+                row(null, null),
+                row(1, Date.valueOf("2020-08-04")),
+                row(2, Date.valueOf("2020-08-04")));
+        onPresto().executeQuery("DROP TABLE hive.default.test_select_drop");
+        assertThat(() -> query("SELECT * FROM iceberg.default.test_select_drop"))
+                .failsWithMessage("Table 'iceberg.default.test_select_drop' does not exist");
+    }
+
+    @Test(groups = {HIVE_REDIRECTION_TO_ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testInsert()
+    {
+        onPresto().executeQuery(format(CREATE_ICEBERG_TABLE_TEAMPLATE, "iceberg.default.test_insert"));
+        onPresto().executeQuery(format(INSERT_TEMPLATE, "hive.default.test_insert"));
+        assertThat(onPresto().executeQuery("SELECT * FROM iceberg.default.test_insert")).containsOnly(EXPECTED_ROWS);
+        assertThat(onPresto().executeQuery("SELECT * FROM hive.default.test_insert")).containsOnly(EXPECTED_ROWS);
+        onPresto().executeQuery("DROP TABLE iceberg.default.test_insert");
+    }
+
+    @Test(groups = {HIVE_REDIRECTION_TO_ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testDescribe()
+    {
+        onPresto().executeQuery(format(CREATE_ICEBERG_TABLE_TEAMPLATE, "iceberg.default.test_describe"));
+        assertThat(onPresto().executeQuery("DESCRIBE hive.default.test_describe"))
```

Comment:
Perhaps unrelated to this PR, but what happens if we create with an Iceberg type like `TIME` that Hive doesn't support? (`UUID` would be another example but @rdblue says that type should be removed)

Reply:
I believe we would be able to create such a table and query it through Iceberg Connector, but unable to query it through Hive Connector. (https://github.com/prestosql/presto/blob/6e43e24194f7d3cb61d5222dd7c9ccafdc4f4873/presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/HiveTableOperations.java and https://github.com/apache/iceberg/blob/c36740be1eed8f1773b9f48bcf498f56dc784d84/hive/src/main/java/org/apache/iceberg/hive/HiveTypeConverter.java)
Since this PR's implementation redirects the query to Iceberg Connector entirely, people can query it as `hive.default.test_describe`, too